python/BioSimSpace/FreeEnergy/_atm.py

```diff
@@ -0,0 +1,1393 @@
+                colour,
+                0.7,
+            )
+            ngl.shape.add_sphere(
+                [coord2.x().value(), coord2.y().value(), coord2.z().value()],
+                colour,
+                0.7,
+            )
+        if system is not None:
+            del local_s
+        return ngl
+
+    @staticmethod
+    def run(
+        system,
+        protocol,
+        platform="CPU",
+        work_dir=None,
+        setup_only=False,
+        property_map={},
+    ):
+        """Run the AToM production simulation(s).
+
+        Parameters
+        ----------
+        system : :class:`System <BioSimSpace._SireWrappers.System>`
+            A prepared AToM system.
+
+        protocol : :class:`Protocol <BioSimSpace.Protocol.AToMProduction>`
+            A protocol object that defines the AToM protocol.
+
+        platform : str
+            The platform for the simulation.
+
+        work_dir : str
+            The working directory for the simulation.
+
+        setup_only : bool
+            Whether to only support simulation setup. If True, then no
+            simulation processes objects will be created, only the directory
+            hierarchy and input files to run a simulation externally. This
+            can be useful when you don't intend to use BioSimSpace to run
+            the simulation. Note that a 'work_dir' must also be specified.
+
+        property_map : dict
+            A dictionary that maps system "properties" to their user defined
+            values. This allows the user to refer to properties with their
+            own naming scheme, e.g. { "charge" : "my-charge" }
+
+        """
+        runner = _relativeATM(
+            system,
+            protocol,
+            platform,
+            property_map=property_map,
+            work_dir=work_dir,
+            setup_only=setup_only,
+        )
+        runner.run()
+        if setup_only:
+            return None
+        else:
+            return runner._runner
+
+    @staticmethod
+    def analyse(
+        work_dir,
+        method="UWHAM",
+        ignore_lower=0,
+        ignore_upper=None,
+        inflex_indices=None,
+    ):
+        """Analyse the AToM simulation.
+
+        Parameters
+        ----------
+        work_dir : str
+            The working directory where the AToM simulation is located.
+        method : str
+            The method to use for the analysis. Currently only UWHAM is supported.
+        ignore_lower : int
+            Ignore the first N samples when analysing.
+        inflex_indices : [int]
+            The indices at which the direction changes. For example, if direction=[1,1,-1,-1],
+            then inflex_indices=[1,2].
+            If None, the inflexion point will be found automatically.
+
+        Returns
+        -------
+        ddg : float
+            The free energy difference between the two ligands.
+        ddg_err : float
+            The error in the free energy difference.
```

Comment:
Return `BioSimSpace.Types.Energy` types.  I guess this might also need to return a overlap matrix, conditional on the `method`?

Reply:
As it stands, since `UWHAM` is the only functioning analysis, I've not added any additional returns from `analysis`, if we are able to get `MBAR` working in the future this can be revisited